### DISCUSSION TITLE
Could not build in gastby cloud with a library that is outside the gatsby app folder

### DISCUSSION THREAD

**Discussion by xiongemi** [gatsbyjs/gatsby#31254] [Help]
## Description

Describe the issue that you're seeing.

- Got a nx monorepo that has a gatsby app and it imports from a library that is outside the app folder; however, when building inside the gatsby cloud, it has an error that any npm packages used in the library are not found.

https://www.gatsbyjs.com/dashboard/77216553-029f-44e7-8134-9b6c85710e0a/sites/ac5a1f5e-5ae5-4f58-8c0c-8e6a9476e6a3/builds/abc47797-fd78-46cd-83fc-c9e1095493d5/details#errors

### Steps to reproduce

Clear steps describing how to reproduce the issue. **Please** link to a reproduction, this makes your issue _much_ easier to diagnose (seriously).

It uses this repo:
https://github.com/xiongemi/nx-gatsby-blogs

In the gatsby cloud settings:
- Production branch: master
- Base directory: apps/blog-medium

I think when installing dependencies, it goes to folder `/usr/src/app/www/apps/blog-medium/node_modules` instead of root level.

### Expected result

What should happen?

when doing the npm install, it should go to the root level node_mdoules.

### Actual result

What happened.

Got this error "Generating JavaScript bundles failed at libs/ui/src/lib/author-details/author-details.tsx" with error like
```
ERROR Generating JavaScript bundles failed

18:00:00 PM:
Can't resolve '@material-ui/core' in '/usr/src/app/www/libs/ui/src/lib/author-details'

18:00:00 PM:
If you're trying to use a package make sure that '@material-ui/core' is installed. If you're trying to use a local file make sure that the path is correct.
```

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here. Also name any `flags` you use inside `gatsby-config.js`.

```
  System:
    OS: macOS 11.2.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.12.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.8.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 90.0.4430.93
    Safari: 14.0.3
  npmPackages:
    gatsby: 3.2.1 => 3.2.1 
    gatsby-image: 3.2.0 => 3.2.0 
    gatsby-plugin-env-variables: ^2.1.0 => 2.1.0 
    gatsby-plugin-layout: ^2.3.0 => 2.3.0 
    gatsby-plugin-manifest: 3.2.0 => 3.2.0 
    gatsby-plugin-material-ui: ^3.0.1 => 3.0.1 
    gatsby-plugin-offline: 4.2.0 => 4.2.0 
    gatsby-plugin-react-helmet: 4.2.0 => 4.2.0 
    gatsby-plugin-remove-trailing-slashes: ^3.3.0 => 3.3.0 
    gatsby-plugin-sharp: 3.2.0 => 3.2.0 
    gatsby-plugin-svgr: 3.0.0-beta.0 => 3.0.0-beta.0 
    gatsby-plugin-typescript: 3.2.0 => 3.2.0 
    gatsby-source-filesystem: 3.2.0 => 3.2.0 
    gatsby-source-medium: ^3.3.0 => 3.3.0 
    gatsby-source-wordpress: ^5.3.0 => 5.3.0 
    gatsby-transformer-sharp: 3.2.0 => 3.2.0 
  npmGlobalPackages:
    gatsby-cli: 3.3.0

```

**Comment by KyleAMathews** [comment]
How do NX monorepos work? Do you just need to run `npm install` at the root? I'm not sure if we detect NX monorepos atm.

**Reply by xiongemi** [reply]
nx monorepo is single version control. yes, it needs to run npm install at root. i think the quick way to detect it is to see whether there is a nx.json at root.

**Comment by LekoArts** [comment]
For support specific to Gatsby Cloud please head here: https://www.gatsbyjs.com/support/
